infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/environment/OpenShiftCheInstallationLocation.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.workspace.infrastructure.openshift.environment;
+
+import com.google.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * OpenShiftCheInstallationLocation checks the KUBERNETES_NAMESPACE and POD_NAMESPACE environment
+ * variables to determine what namespace Che is installed in. Users should use this class to
+ * retrieve the installation namespace name.
+ *
+ * @author Tom George
+ */
+@Singleton
+public class OpenShiftCheInstallationLocation {
+
+  private static final Logger LOG = LoggerFactory.getLogger(OpenShiftCheInstallationLocation.class);
+
+  private String installationLocationNamespace;
+
+  @Inject(optional = true)
+  @Named("env.KUBERNETES_NAMESPACE")
+  private String kubernetesNamespace = null;
+
+  @Inject(optional = true)
+  @Named("env.POD_NAMESPACE")
+  private String podNamespace = null;
+
+  /** @return The name of the namespace where Che is installed */
+  public String getInstallationLocationNamespace() {
+    if (kubernetesNamespace == null && podNamespace == null) {
+      LOG.info(
+          "Neither KUBERNETES_NAMESPACE nor POD_NAMESPACE is defined.  Unable to determine Che installation location");
+    }
+    return kubernetesNamespace == null ? podNamespace : kubernetesNamespace;
```

Comment:
Maybe a warn if both `podNamespace` and `kubernetesNamespace` are defined but not matching would also make sense? I don't know if this can occur naturally but might be a hard-to-debug issue.

Reply:
in general, we do need to standardize the env var name (use only KUBERNETES_NAMESPACE) and remove usage of `POD_NAMESPACE` e.g. in helm chart https://github.com/eclipse/che/blob/master/deploy/kubernetes/helm/che/templates/deployment.yaml#L38

But we should do it in a separate PRs I beleive after 7.12.0